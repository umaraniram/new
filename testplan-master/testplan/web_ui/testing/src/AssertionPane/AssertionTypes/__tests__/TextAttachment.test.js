/* Unit tests for the TextAttachment component. */
import React from 'react';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";
import moxios from 'moxios';

import TextAttachment from '../TextAttachment.js';

describe('TextAttachment', () => {
  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
    moxios.install();
  });

  afterEach(() => {
    // Resume style injection once test is finished.
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
    moxios.uninstall();
  });

  it('displays loading spinner when mounted', () => {
    const renderedText = shallow(
      <TextAttachment
        src="/var/tmp/attachment.txt"
        file_name="attachment.txt"
      />
    );
    expect(renderedText.state("loading")).toBeTruthy();
    expect(renderedText).toMatchSnapshot();
  });

  it('renders text returned from the backend', () => {
    const text = "testplan\n".repeat(100);
    const renderedText = shallow(
      <TextAttachment
        src="/var/tmp/attachment.txt"
        file_name="attachment.txt"
      />
    );
    renderedText.instance().handleText(text);
    renderedText.update();
    expect(renderedText.state("originalText")).toBe(text + "<newline>");
    expect(renderedText).toMatchSnapshot();
  });

  it('displays error message if API request fails', done => {
    const renderedText = shallow(
      <TextAttachment
        src="/var/tmp/attachment.txt"
        file_name="attachment.txt"
      />
    );
    moxios.wait(() => {
      const request = moxios.requests.mostRecent();
      request.respondWith({
        status: 503,
        message: "Service Unavailable",
      }).then(() => {
        renderedText.update();
        expect(renderedText.state("error")).toBeTruthy();
        expect(renderedText).toMatchSnapshot();
        done();
      });
    });
  });
});

