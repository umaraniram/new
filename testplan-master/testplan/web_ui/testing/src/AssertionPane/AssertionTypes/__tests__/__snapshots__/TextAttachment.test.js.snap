// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAttachment displays error message if API request fails 1`] = `
<div>
  Error fetching attachment
  <div>
    <div>
      <a
        href="/var/tmp/attachment.txt"
      >
        attachment.txt
      </a>
    </div>
    <br />
    <div
      className={null}
    />
    <br />
    <div />
  </div>
</div>
`;

exports[`TextAttachment displays loading spinner when mounted 1`] = `
<div>
  <div
    className="spinner_1hfbsw5"
  >
    <Loader
      color="#123abc"
      css={Object {}}
      height={10}
      heightUnit="px"
      loading={true}
      margin="2px"
      radius={4}
      radiusUnit="px"
      width={3}
      widthUnit="px"
    />
  </div>
</div>
`;

exports[`TextAttachment renders text returned from the backend 1`] = `
<div>
  <div>
    <div>
      <a
        href="/var/tmp/attachment.txt"
      >
        attachment.txt
      </a>
    </div>
    <br />
    <div
      className={null}
    >
      <SyntaxHighlighter
        showLineNumbers={true}
        startingLineNumber={81}
      >
        ...
testplan
testplan
testplan
testplan
testplan
testplan
testplan
testplan
testplan
testplan
testplan
testplan
testplan
testplan
testplan
testplan
testplan
testplan
testplan
&lt;newline&gt;

      </SyntaxHighlighter>
    </div>
    <br />
    <div>
      <button
        onClick={[Function]}
      >
        Expand
      </button>
    </div>
  </div>
</div>
`;
